health-on-time/src/App.js

```diff
@@ -80,24 +81,52 @@ function App() {
     }
   }, []);
 
+  /********  
+  // Rewrote below to include using previous state in functional setState and 
+  // to update prescriptions along with the medications change
+  */
   let deleteMedication = (medicationID) => {
     console.log(medicationID, "med id");
     fetch(`${BASE_URL}medications/${medicationID}`, {
       method: "DELETE",
     })
       .then((r) => r.json())
       .then((deletedMedication) => {
-        let copyOfMeds = medications.filter((med) => {
-          return med.id !== medicationID;
-        });
-        setMedications(copyOfMeds);
+        setMedications(prevMeds => {
+          return prevMeds.filter(med => med.id !== medicationID);
+        })
+        setPrescriptions(prevPrescriptions => {
+          return prevPrescriptions.filter(prescr => prescr.medication.id !== medicationID);
+        })
       });
   };
 
+  const handleNewPrescription = (newPrescriptionObj) => {
+    fetch(`${BASE_URL}prescriptions`, {
+      method: "POST",
+      headers: {
+        "content-type": "application/json",
+        Authorization: `Bearer ${localStorage.token}`,
+      },
+      body: JSON.stringify(newPrescriptionObj),
+    })
+      .then((r) => r.json())
+      .then(data => { // update locally w/ setPrescriptions
+        setPrescriptions(prevPrescriptions => [...prevPrescriptions, data.prescription]);
+        setMedications(prevMeds => [...prevMeds, data.prescription.medication]);
+      }) // can check created object
+      .catch(error => {
+        console.log(error.name + ": " + error.message);
+        throw error; // or return error message?
+      });
+    // PROBLEM: Medications not updated
+  } // end function handleNewPrescription
+
   return (
     <div className="main-container">
       <Navbar loggedIn={loggedIn} handleLogout={handleLogout} />
-      <div id={loggedIn && "wrapper"}>
+      {/* <div id={loggedIn && "wrapper"}>  React complains/warns about recieving id=false} Lewis */}
+      <div id={loggedIn ? "wrapper" : undefined}> 
         {loggedIn && <Sidebar prescriptions={prescriptions} />}
         <div className="display">
           <UserContext.Provider
```

Comment:
No  handleNewPrescription ?

Reply:
a) I'm confused about your comment. Where should handleNewPrescription be? 
b) I'm confused about what line 129 (100 in red) is actually doing. All I did was add the : undefined because React didn't like that id could get set to false when id should be non-boolean. I guess if logged in is true, it applies the wrapper styles to the outer div. But I don't understand it totally.
c) Line 130: I don't know why Sidebar gets prescriptions prop. I removed it now for this PR

Reply:
![image](https://user-images.githubusercontent.com/71042305/111223451-c8809c00-85a2-11eb-93bb-698befdca55d.png)

Reply:
Thanks for fixing the bug.
I see that you added `handleNewPrescriptions` & ect in the provider, that's what I wanted to say.